### PR TITLE
Add static Myself Name

### PR DESCRIPTION
**Pull request by xiaobill8**
- Add support for Myself parsing
- Update User Guide

### CODE REVIEW

**Review by seetohjinwei — APPROVED** [review]
LGTM!

File: docs/UserGuide.md [hunk 1]
```diff
@@ -106,43 +109,60 @@ Examples:
 
 Adds a task of type todo into the task list.
 
-Format:  `task todo m/ASSIGNOR d/DESCRIPTION`
+**Assigned by** Format:  `task todo m/NAME d/DESCRIPTION`
+
+**Assigned to** Format:  `task todo o/Name d/DESCRIPTION`
```
Comment (Nephelite) [inline docs/UserGuide.md:114]: Name probably should be all caps

File: docs/UserGuide.md [hunk 2]
```diff
@@ -384,12 +404,12 @@ The following date formats are accepted:
...
-| **Add Deadline: Assignee**    | `task deadline o/ASSIGNEE d/DESCRIPTION t/DATE` <br> e.g., `task deadline o/Sam d/Finish the assignment t/Jan 31 2022`                                                   |
-| **Add Event: Assignor**       | `task event m/ASSIGNOR d/DESCRIPTION t/DATE` <br> e.g., `task event m/John d/Finish user guide t/2022-12-31`                                                             |
-| **Add Event: Assignee**       | `task event o/ASSIGNEE d/DESCRIPTION t/DATE` <br> e.g., `task event o/Sam d/Finish the assignment t/Jan 31 2022`                                                         |
+| **Add Todo: Assigned by**        | `task todo m/NAME d/DESCRIPTION` <br> e.g., `task todo m/John d/Finish user guide`                                                                                   |
+| **Add Todo: Assigned to**        | `task todo o/Name d/DESCRIPTION` <br> e.g., `task todo o/Sam d/Finish the assignment`                                                                                |
```
Comment (Nephelite) [inline docs/UserGuide.md:408]: Name should be all caps here too.

File: docs/UserGuide.md [hunk 3]
```diff
@@ -384,12 +404,12 @@ The following date formats are accepted:
...
+| **Add Deadline: Assigned by**    | `task deadline m/NAME d/DESCRIPTION t/DATE` <br> e.g., `task deadline m/John d/Finish user guide t/2022-12-31`                                                       |
+| **Add Deadline: Assigned to**    | `task deadline o/Name d/DESCRIPTION t/DATE` <br> e.g., `task deadline o/Sam d/Finish the assignment t/Jan 31 2022`                                                   |
```
Comment (Nephelite) [inline docs/UserGuide.md:410]: Here too.

File: docs/UserGuide.md [hunk 4]
```diff
@@ -384,12 +404,12 @@ The following date formats are accepted:
...
+| **Add Event: Assigned by**       | `task event m/NAME d/DESCRIPTION t/DATE` <br> e.g., `task event m/John d/Finish user guide t/2022-12-31`                                                             |
+| **Add Event: Assigned to**       | `task event o/Name d/DESCRIPTION t/DATE` <br> e.g., `task event o/Sam d/Finish the assignment t/Jan 31 2022`                                                         |
```
Comment (Nephelite) [inline docs/UserGuide.md:412]: Here too.

**Review by Nephelite — CHANGES_REQUESTED** [review]

**Review by Nephelite — APPROVED** [review]

**Pull request merged**